components/script/dom/webgl_extensions/wrapper.rs

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use core::nonzero::NonZero;
+use dom::bindings::js::{MutNullableJS, Root};
+use dom::bindings::reflector::DomObject;
+use dom::bindings::trace::JSTraceable;
+use dom::webglrenderingcontext::WebGLRenderingContext;
+use heapsize::HeapSizeOf;
+use js::jsapi::JSObject;
+use std::any::Any;
+use super::{WebGLExtension, WebGLExtensionManager};
+
+/// Trait used internally by WebGLExtensionManager to store and
+// handle the different WebGL extensions in a common list.
+pub trait WebGLExtensionWrapper: JSTraceable + HeapSizeOf {
+    fn instance_or_init(&self,
+                        ctx: &WebGLRenderingContext,
+                        manager: &WebGLExtensionManager)
+                        -> NonZero<*mut JSObject>;
+    fn is_supported(&self, &WebGLExtensionManager) -> bool;
+    fn enable(&self, manager: &WebGLExtensionManager);
+    fn name(&self) -> &'static str;
+    fn as_any<'a>(&'a self) -> &'a Any;
```

Comment:
I tried to use directly the Any trait instead of adding the method but I couldn't. Here is an explanation: http://stackoverflow.com/a/33687996

Yes, the lifetime declaration was unneeded.